airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/VariableInputFieldForm.tsx

```diff
@@ -0,0 +1,82 @@
+import { useField } from "formik";
+import { FormattedMessage } from "react-intl";
+import { useAsync, useEffectOnce } from "react-use";
+import * as yup from "yup";
+
+import { Button, ModalBody, ModalFooter } from "components";
+
+import { FormGroupItem, FormObjectArrayItem } from "core/form/types";
+
+import { useServiceForm } from "../../serviceFormContext";
+import { FormSection } from "./FormSection";
+interface VariableInputFormProps {
+  formField: FormObjectArrayItem;
+  path: string;
+  item?: unknown;
+  disabled?: boolean;
+  onDone: (value: unknown) => void;
+  onCancel: () => void;
+}
+
+export const VariableInputFieldForm: React.FC<VariableInputFormProps> = ({
+  formField,
+  path,
+  item,
+  disabled,
+  onDone,
+  onCancel,
+}) => {
+  const hiddenPath = `__hidden_${path}`;
+  const [field, , fieldHelper] = useField(hiddenPath);
+  const { validationSchema } = useServiceForm();
+
+  const { value: isValid } = useAsync(
+    async (): Promise<boolean> => yup.reach(validationSchema, path).isValid(field.value),
+    [field.value, path, validationSchema]
+  );
+
+  useEffectOnce(() => {
+    // Find initial values if not editing item
+    const initialValue = item
+      ? {}
+      : (formField.properties as FormGroupItem).properties.reduce((acc, item) => {
+          if (item._type === "formItem" && item.default) {
```

Comment:
Is the check for `formItem` needed here, or couldn't we set the `default` value for all fields (no matter their type, i.e. even if they'd be complex types)?

Reply:
for posterity from our offline discussion: Won't work because this is a typing question and `.default` only exists on `formItem` types.